projects/extension/src/background/ConnectionManager.ts

```diff
@@ -342,8 +305,11 @@ export class ConnectionManager<SandboxId> {
         }
 
         // Refuse the chain addition for invalid well-known chain names.
-        if (message.type === "add-well-known-chain") {
-          if (!this.#wellKnownChains.has(message.chainName)) {
+        if (
+          message.type === "add-well-known-chain" ||
+          message.type === "add-well-known-chain-with-db"
+        ) {
+          if (!this.#wellKnownChainSpecs.has(message.chainName)) {
```

Comment:
Since [keys of wellKnownChains](https://github.com/paritytech/substrate-connect/blob/main/projects/extension/src/background/index.ts#L21-L26) are all `.id` for better readability and less confusion lets use the chainId ([for well known chains `chainName === chainId` anyways](https://github.com/paritytech/substrate-connect/pull/1047/files#diff-4de63d371461c5b51ea9a173eccc4c3b114374baa9388b564a3033e112f58924R221-R222))
```suggestion
          if (!this.#wellKnownChainSpecs.has(message.chainId)) {
```

Reply:
That's wrong. The message can come from the connect package, where the `chainId` is assigned randomly.

Reply:
> That's wrong. The message can come from the connect package, where the `chainId` is assigned randomly.

Found it:
`const chainId = getRandomChainId()`
I think we should alter the [part in connect](https://github.com/paritytech/substrate-connect/blob/main/packages/connect/src/connector/extension.ts#L97-L101) when its a well-known-chain to assign to the chainId the name (as we do in extension as well), or randomise the ID in the extension. Its not about the change I propos but IMO its good to have a common approach in both cases. Opinions?

Reply:
That'd be wrong as well. Multiple libraries based on top of connect and running at the same time need to be able to create each their own connection to a well-known chain, and thus the `chainId` needs to be different to distinguish these multiple connections.

Reply:
Thank you for the feedback. Then Its good to be merged.. I mean both comments are just "comments for possible improvement" and PR is already approved.